Source/WebCore/Modules/identity/DigitalCredential.cpp

```diff
@@ -105,13 +130,25 @@ void DigitalCredential::discoverFromExternalSource(const Document& document, Cre
 
 #if HAVE(DIGITAL_CREDENTIALS_UI)
     DigitalCredentialsRequestData requestData;
-    requestData.options = options.digital.value();
+    for (auto& request : options.digital->requests) {
+        auto resultOrException = jsToCredentialRequest(document, request);
+        if (resultOrException.hasException()) {
+            promise.reject(resultOrException.releaseException());
+            return;
+        }
+
+        DigitalCredentialRequestTypes credentialVariant = resultOrException.releaseReturnValue();
+        std::visit([&](auto& credential) {
+            requestData.requests.append(credential);
+        },
+            credentialVariant);
+    }
     requestData.topOrigin = document.protectedTopOrigin()->data().isolatedCopy();
     requestData.documentOrigin = document.protectedSecurityOrigin()->data().isolatedCopy();
-
-    Ref coordinator = document.protectedFrame()->protectedPage()->credentialRequestCoordinator();
-    if (!coordinator->presentPicker(WTFMove(promise), WTFMove(requestData), options.signal))
-        LOG(DigitalCredentials, "Failed to present the credential picker.");
+    document.protectedFrame()
+        ->protectedPage()
+        ->credentialRequestCoordinator()
+        .presentPicker(WTFMove(promise), WTFMove(requestData), options.signal);
```

Comment:
Hmm, we don't usually wrap chained calls like this. No strong preference though.

Reply:
Too jQuery? hehe...